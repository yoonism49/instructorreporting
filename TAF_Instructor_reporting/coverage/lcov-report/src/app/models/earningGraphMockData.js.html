<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/app/models/earningGraphMockData.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/app/models</a> earningGraphMockData.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26</td><td class="line-coverage quiet"><span class="cline-any cline-yes">14x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const earningGraphMockData = {
  labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
  datasets: [
    {
      label: 'My First dataset',
      fillColor: 'rgba(220,220,220,0.2)',
      strokeColor: 'rgba(220,220,220,1)',
      pointColor: 'rgba(220,220,220,1)',
      pointStrokeColor: '#fff',
      pointHighlightFill: '#fff',
      pointHighlightStroke: 'rgba(220,220,220,1)',
      data: [65, 59, 80, 81, 56, 55, 40]
    },
    {
      label: 'My Second dataset',
      fillColor: 'rgba(151,187,205,0.2)',
      strokeColor: 'rgba(151,187,205,1)',
      pointColor: 'rgba(151,187,205,1)',
      pointStrokeColor: '#fff',
      pointHighlightFill: '#fff',
      pointHighlightStroke: 'rgba(151,187,205,1)',
      data: [28, 48, 40, 19, 86, 27, 90]
    }
  ]
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 02 2018 06:20:09 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
