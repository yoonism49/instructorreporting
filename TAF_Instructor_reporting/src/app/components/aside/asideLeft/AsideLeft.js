// @flow weak

/* eslint no-console: 0 */
import React      from 'react';
import PropTypes  from 'prop-types';
import cx         from 'classnames';
import UserPanel  from './userPanel/UserPanel';
import SearchForm from './searchForm/SearchForm';
import Horloge    from '../../horloge/Horloge';
import Menu       from './menu/Menu';


const AsideLeft = ({
  connectionStatus,
  userIsConnected,
  username,
  helloWord,
  userPicture,
  showPicture,
  isAnimated,
  isCollapsed,
  currentView,
  sideMenu
}) => (
  <aside 
    className={
      cx({
        'no-print':          true,
        'left-side':         true,
        'aside-left--fixed': true,
        'sidebar-offcanvas': true,
        'sidebar-animated':  isAnimated,
        'collapse-left':     isCollapsed
      })
    }
    // add overflow to left sidebar:
    style={{
      height:   '100%',
      overflow: 'auto',
      position: 'fixed'
    }}
  >
    <section className="sidebar">
      <UserPanel
        hello={helloWord}
        username={username}
        connectionStatus={connectionStatus}
        online={userIsConnected}
        userPicture={userPicture}
        showUserPicture={showPicture}
      />
      <Horloge />

      <SearchForm
        onSearchSubmit={(value) => console.log('searching: ', value)}
      />
      {
        sideMenu.map(
          ({id, group, menus}, menuIdx) => {
            return (
              <Menu
                key={menuIdx}
                initialCollapseState={menuIdx === 0 ? false : null}
                headerTitle={group}
                headerBackColor="#283744"
                activeView={currentView}
                views={menus}
                id={id}
              />
            );
          }
        )
      }
    </section>
  </aside>
);

AsideLeft.propTypes = {
  isAnimated:   PropTypes.bool,
  isCollapsed:  PropTypes.bool,
  sideMenu:     PropTypes.arrayOf(
    PropTypes.shape({
      id:     PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
      group:  PropTypes.string.isRequired,
      menus:  PropTypes.arrayOf(
        PropTypes.shape({
          name:       PropTypes.string.isRequired,
          linkTo:     PropTypes.string.isRequired,
          faIconName: PropTypes.string.isRequired
        })
      )
    })
  ).isRequired,
  currentView:      PropTypes.string,
  connectionStatus: PropTypes.shape({
    online:         PropTypes.string,
    disconnected:   PropTypes.string
  }),
  userIsConnected:  PropTypes.bool,
  username:         PropTypes.string,
  userPicture:      PropTypes.any,
  showPicture:      PropTypes.bool,
  helloWord:        PropTypes.string
};

AsideLeft.defaultProps = {
  isAnimated:   false,
  isCollapsed:  false
};

export default AsideLeft;
