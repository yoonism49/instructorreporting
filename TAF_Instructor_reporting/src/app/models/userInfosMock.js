export const userInfosMockData = {
  login: 'johnDoeIsNotAffraid',
  firstname: 'John',
  lastname: 'Doe',
  picture:  require('../img/user.jpg'), // or from an url: 'https://placeimg.com/120/120/people', // or from a relative path (NOTE: this path like public/.. may not be availaible when in dev hot reload!) './public/img/user.jpg',
  showPicture: true,
  token: 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJkZW1vIiwiaWF0IjoxNTAyMzA3MzU0LCJleHAiOjE3MjMyMzIxNTQsImF1ZCI6ImRlbW8tZGVtbyIsInN1YiI6ImRlbW8iLCJHaXZlbk5hbWUiOiJKb2huIiwiU3VybmFtZSI6IkRvZSIsIkVtYWlsIjoiam9obi5kb2VAZXhhbXBsZS5jb20iLCJSb2xlIjpbIlN1cGVyIGNvb2wgZGV2IiwibWFnaWMgbWFrZXIiXX0.6FjgLCypaqmRp4tDjg_idVKIzQw16e-z_rjA3R94IqQ'
};
export default userInfosMockData;
