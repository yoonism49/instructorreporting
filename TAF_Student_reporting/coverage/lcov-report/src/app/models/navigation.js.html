<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/app/models/navigation.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/app/models</a> navigation.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>1/1</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144</td><td class="line-coverage quiet"><span class="cline-any cline-yes">16x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export const navigation = {
  brand:      'TAF',
  leftLinks:  [],
  rightLinks: [
    {
      label:      'Home',
      link:       '/',
      view:       'home',
      isRouteBtn: true
    },
    {
      label:      'About',
      link:       '/about',
      view:       'about',
      isRouteBtn: true
    }
  ],
  sideMenu: [
    // group menu #1
    {
      id: 1,
      group: 'Dashboard  ',
      menus: [
        {
          name: 'Dashboard preview',
          linkTo: '/',
          faIconName: 'fa-eye'
        },
        {
          name: 'StatsCard',
          linkTo: '/Dashboard/statsCard',
          faIconName: 'fa-check-square-o'
        },
        {
          name: 'EarningGraph',
          linkTo: '/Dashboard/earningGraph',
          faIconName: 'fa-area-chart'
        },
        {
          name: 'Notifications',
          linkTo: '/Dashboard/notifications',
          faIconName: 'fa-bell'
        },
        {
          name: 'Work progress',
          linkTo: '/Dashboard/workProgress',
          faIconName: 'fa-briefcase'
        },
        {
          name: 'Twitter feed',
          linkTo: '/Dashboard/twitterFeed',
          faIconName: 'fa-twitter'
        },
        {
          name: 'Team Mates',
          linkTo: '/Dashboard/teamMates',
          faIconName: 'fa-user'
        },
        {
          name: 'Todo list',
          linkTo: '/Dashboard/todoList',
          faIconName: 'fa-check'
        }
      ]
    },
    // group menu #2
    {
      id: 2,
      group: 'General',
      menus: [
        {
          name: 'General preview',
          linkTo: '/general',
          faIconName: 'fa-eye'
        },
        {
          name: 'Breadcrumb',
          linkTo: '/general/breadcrumb',
          faIconName: 'fa-bars'
        },
        {
          name: 'Stat',
          linkTo: '/general/stat',
          faIconName: 'fa-bar-chart'
        },
        {
          name: 'Basic progress bars',
          linkTo: '/general/basicProgressBars',
          faIconName: 'fa-tasks'
        },
        {
          name: 'Tab panels',
          linkTo: '/general/tabPanels',
          faIconName: 'fa-columns'
        },
        {
          name: 'Striped progress bar',
          linkTo: '/general/stripedProgressBars',
          faIconName: 'fa-tasks'
        },
        {
          name: 'Alerts',
          linkTo: '/general/alerts',
          faIconName: 'fa-exclamation-triangle'
        },
        {
          name: 'Pagination',
          linkTo: '/general/pagination',
          faIconName: 'fa-sort'
        },
        {
          name: 'Default buttons',
          linkTo: '/general/defaultButtons',
          faIconName: 'fa-hand-o-up'
        }
      ]
    },
    // group menu #3
    {
      id: 3,
      group: 'BasicElements',
      menus: [
        {
          name: 'Basic Elements preview',
          linkTo: '/basicElements',
          faIconName: 'fa-eye'
        }
      ]
    },
    // group menu #4
    {
      id: 4,
      group: 'SimpleTables',
      menus: [
        {
          name: 'Simple tables preview',
          linkTo: '/simpleTables',
          faIconName: 'fa-eye'
        }
      ]
    }
  ]
};
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Feb 02 2018 06:20:09 GMT+0100 (CET)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
