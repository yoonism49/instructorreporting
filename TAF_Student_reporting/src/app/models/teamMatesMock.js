export const teamMatesMock = [
  {
    picture: '/public/img/26115.jpg',
    firstname: 'Damon',
    lastname: 'Parker',
    profile: 'Admin',
    profileColor: 'danger'
  },
  {
    picture: '/public/img/26115.jpg',
    firstname: 'Joe',
    lastname: 'Waston',
    profile: 'Member',
    profileColor: 'warning'
  },
  {
    picture: '/public/img/26115.jpg',
    firstname: 'Jannie',
    lastname: 'Davis',
    profile: 'Editor',
    profileColor: 'warning'
  },
  {
    picture: '/public/img/26115.jpg',
    firstname: 'Emma',
    lastname: 'Welson',
    profile: 'Editor',
    profileColor: 'success'
  },
  {
    picture: '/public/img/26115.jpg',
    firstname: 'Emma',
    lastname: 'Welson',
    profile: 'Editor',
    profileColor: 'info'
  }
];
