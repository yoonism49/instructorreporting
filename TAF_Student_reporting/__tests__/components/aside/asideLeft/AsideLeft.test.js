'use strict';

import React            from 'react';
import AsideLeft        from '../../../../src/app/components/aside/asideLeft/AsideLeft';
import renderer         from 'react-test-renderer'; // needed both for snpashot testing but also to prevent errors from enzyme
import { navigation }   from '../../../../src/app/models/navigation';
import { MemoryRouter } from 'react-router';

// mock Horloge since snapshot would be different at each run
jest.mock(
  '../../../../src/app/components/horloge/Horloge',
  () => 'Horloge'
);

const sideMenus = navigation.sideMenu;

describe('AsideLeft component', () => {
  function createNodeMock() {
    return {
      focus: () => {}
    };
  }
  it('renders as expected', () => {
    // we need MemoryRouter to set an history
    // for children that uses Link from react-router
    const component = renderer.create(
      <div>
        <MemoryRouter>
          <AsideLeft
            isAnimated
            isCollapsed
            sideMenu={sideMenus}
            currentView={sideMenus[0].menus[0].name}
          />
        </MemoryRouter>
      </div>,
      { createNodeMock }
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
