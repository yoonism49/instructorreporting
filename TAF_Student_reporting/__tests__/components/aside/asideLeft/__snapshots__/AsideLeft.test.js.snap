// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AsideLeft component renders as expected 1`] = `
<div>
  <aside
    className="no-print left-side aside-left--fixed sidebar-offcanvas sidebar-animated collapse-left"
    style={
      Object {
        "height": "100%",
        "overflow": "auto",
        "position": "fixed",
      }
    }
  >
    <section
      className="sidebar"
    >
      <div
        className="user-panel"
      >
        <div
          className="pull-left image"
        >
          <img
            alt="User Image"
            className="img-circle"
            src={undefined}
          />
        </div>
        <div
          className="pull-left info"
        >
          <p>
            Hello Jane
          </p>
          <a
            href="#"
          >
            <i
              className="fa fa-circle text-default"
            />
            Disconnected
          </a>
        </div>
      </div>
      <Horloge />
      <form
        className="sidebar-form"
        onKeyPress={[Function]}
      >
        <div
          className="input-group"
        >
          <input
            className="form-control"
            name="searchinput"
            placeholder="Search..."
            type="text"
          />
          <span
            className="input-group-btn"
          >
            <button
              className="btn btn-flat"
              id="search-btn"
              name="search"
              onClick={[Function]}
            >
              <i
                className="fa fa-search"
              />
            </button>
          </span>
        </div>
      </form>
      <div>
        <ul
          className="list-inline sidebar-menu__group-container"
          onClick={[Function]}
        >
          <li
            style={
              Object {
                "backgroundColor": "#283744",
                "width": "100%",
              }
            }
          >
            <a
              className="btn"
              style={
                Object {
                  "paddinLeft": "0px !important",
                  "width": "100%",
                }
              }
            >
              <span
                className="pull-left sidebar-menu__group-title"
              >
                Dashboard  
              </span>
              <i
                aria-hidden="true"
                className="fa fa-angle-up fa-1x pull-right"
                style={
                  Object {
                    "WebkitTransform": "rotate(180deg)",
                    "transform": "rotate(180deg)",
                  }
                }
              />
            </a>
          </li>
        </ul>
        <div
          className="ReactCollapse--collapse"
          style={
            Object {
              "height": 0,
              "overflow": "hidden",
            }
          }
        >
          <div
            className="ReactCollapse--content"
          >
            <ul
              className="sidebar-menu sidebar-menu__marginTop"
            >
              <li
                className="active"
              >
                <a
                  href="/"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-eye"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    Dashboard preview
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/Dashboard/statsCard"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-check-square-o"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    StatsCard
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/Dashboard/earningGraph"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-area-chart"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    EarningGraph
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/Dashboard/notifications"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-bell"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    Notifications
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/Dashboard/workProgress"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-briefcase"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    Work progress
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/Dashboard/twitterFeed"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-twitter"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    Twitter feed
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/Dashboard/teamMates"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-user"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    Team Mates
                  </span>
                </a>
              </li>
              <li
                className=""
              >
                <a
                  href="/Dashboard/todoList"
                  onClick={[Function]}
                >
                  <i
                    className="fa fa-check"
                    style={
                      Object {
                        "marginLeft": "10px",
                      }
                    }
                  />
                  <span
                    style={
                      Object {
                        "fontSize": "13px",
                      }
                    }
                  >
                    Todo list
                  </span>
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div>
        <ul
          className="list-inline sidebar-menu__group-container"
          onClick={[Function]}
        >
          <li
            style={
              Object {
                "backgroundColor": "#283744",
                "width": "100%",
              }
            }
          >
            <a
              className="btn"
              style={
                Object {
                  "paddinLeft": "0px !important",
                  "width": "100%",
                }
              }
            >
              <span
                className="pull-left sidebar-menu__group-title"
              >
                General
              </span>
              <i
                aria-hidden="true"
                className="fa fa-angle-up fa-1x pull-right"
                style={
                  Object {
                    "WebkitTransform": "rotate(180deg)",
                    "transform": "rotate(180deg)",
                  }
                }
              />
            </a>
          </li>
        </ul>
      </div>
      <div>
        <ul
          className="list-inline sidebar-menu__group-container"
          onClick={[Function]}
        >
          <li
            style={
              Object {
                "backgroundColor": "#283744",
                "width": "100%",
              }
            }
          >
            <a
              className="btn"
              style={
                Object {
                  "paddinLeft": "0px !important",
                  "width": "100%",
                }
              }
            >
              <span
                className="pull-left sidebar-menu__group-title"
              >
                BasicElements
              </span>
              <i
                aria-hidden="true"
                className="fa fa-angle-up fa-1x pull-right"
                style={
                  Object {
                    "WebkitTransform": "rotate(180deg)",
                    "transform": "rotate(180deg)",
                  }
                }
              />
            </a>
          </li>
        </ul>
      </div>
      <div>
        <ul
          className="list-inline sidebar-menu__group-container"
          onClick={[Function]}
        >
          <li
            style={
              Object {
                "backgroundColor": "#283744",
                "width": "100%",
              }
            }
          >
            <a
              className="btn"
              style={
                Object {
                  "paddinLeft": "0px !important",
                  "width": "100%",
                }
              }
            >
              <span
                className="pull-left sidebar-menu__group-title"
              >
                SimpleTables
              </span>
              <i
                aria-hidden="true"
                className="fa fa-angle-up fa-1x pull-right"
                style={
                  Object {
                    "WebkitTransform": "rotate(180deg)",
                    "transform": "rotate(180deg)",
                  }
                }
              />
            </a>
          </li>
        </ul>
      </div>
    </section>
  </aside>
</div>
`;
